### ISSUE TITLE
Goat Horn Sounds Not Heard on Bedrock Edition

### ISSUE THREAD

**Issue by ExtremeFiretop** [GeyserMC/Geyser#3082]
### Describe the bug

When using Goat Horns with Minecraft Bedrock Edition version 1.19 and a Java Paper Server version 1.19; the player that sounded the Goat horn sounds do not hear them correctly.

### To Reproduce

1. Download the latest build of Paper (Currently version 33)
2. Download the latest build of Geyser (Currently version Dev 1126)
3. No other plugins except Geyser and Floodgate.
4. Connect to the Java server using an Bedrock and attempt to use the goat horns. 

To clarify; It only seems to be an issue with bedrock players hearing their own horn.

### Expected behaviour

For the horns to make the expected sound as in Java edition and reset the timer so bedrock players can hear their own horns. 

### Screenshots / Videos

https://drive.google.com/file/d/1MX7D4MHvLnHwolP_pwgqZmhLPFvp2hkD/view

### Server Version and Plugins

[16:50:29 INFO]: This server is running Paper version git-Paper-32 (MC: 1.19) (Implementing API version 1.19-R0.1-SNAPSHOT) (Git: 4c9317c)
You are running the latest version

### Geyser Dump

https://dump.geysermc.org/AKirXnzkEHtjQzDGdz18yX2HqGzQhQdN

### Geyser Version

Geyser version 2.0.4-SNAPSHOT (git-master-08241b8)

### Minecraft: Bedrock Edition Device/Version

Minecraft Bedrock 1.19 on Xbox

**Comment by Firetop** [comment]
To clarify; if a Bedrock player presses and holds the horn to use it for an extended amount of time, it seems to work for a Java player, but not a bedrock player. 

As in the bedrock player will never hear the sound, even if a Java player eventually hears it. 
On Java you just click once and it works as expected. 

On bedrock with Geyser, the player needs to press and hold for an extended amount of time (unknown since the horn's reset bar doesn't work) until a Java player might be able to hear it, but the bedrock players will never hear it and never know it went off since there is no sound and no reset bar.

**Comment by Firetop** [comment]
I updated the bug report a few times until it was as accurate as I could make it. Hopefully it didn't cause too much confusion. I added a video to assist with the clarification that the bedrock player will never hear their horn, only the Java player will. If I mute Java the sound no longer happens.

**Comment by Camotoy** [comment]
The next Geyser update will now show the goat horn cooldown. I know why the sound isn't playing (it's played on the client for the player who played it, in Java Edition, and Bedrock needs the sound forwarded to it), but to add that to the codebase at this time is messy. It'll likely get a refactor soon.

**Comment by Firetop** [comment]
> The next Geyser update will now show the goat horn cooldown. I know why the sound isn't playing (it's played on the client for the player who played it, in Java Edition, and Bedrock needs the sound forwarded to it), but to add that to the codebase at this time is messy. It'll likely get a refactor soon.

Thank you for looking into it; once I realized it was only the bedrock player that used the horn which couldn't hear it; it naturally becomes much lower priority than anything else your working on, but I still appreciate the time to investigate it!

**Comment by Jacob439** [comment]
I wrote a plugin to patch this issue which can be found here: https://www.spigotmc.org/resources/goat-horn-bedrock.116914/

I did not make a PR because I don't think this is a very clean solution, but here is my code if someone wants to write a PR based on what I wrote: https://github.com/Jacob439/GoatHornBedrock

Details of my implementation:
1. The music instrument metadata of any goat horn that is added to a player's inventory using an `EntityPickupItemEvent` is saved in item metadata
2. On a `PlayerInteractEvent` I check if the item is a goat horn
3. I reference my previously saved metadata to play the corresponding sound
